src/main/cpp/Shooter.cpp

```diff
@@ -229,11 +229,23 @@ Shooter::Aim(){
     frc::SmartDashboard::PutNumber("hood angle", m_hood.GetSelectedSensorPosition());
     
     // Set Hood Position
-    // if(m_hood.GetSupplyCurrent() >= ShooterConstants::zeroingcurrent){
-    //     m_hood.Set(ControlMode::PercentOutput, 0.0);
-    // } else {
-    m_hood.Set(ControlMode::Position, m_angle);
-    // }
+    if(m_hood.GetSupplyCurrent() >= ShooterConstants::zeroingcurrent){
+         m_hood.Set(ControlMode::PercentOutput, 0.0);
+         return;
+    } 
+    else if(m_hood.GetSelectedSensorPosition() > ShooterConstants::hoodMax &&
+        m_angle >= ShooterConstants::hoodMax){
+        m_hood.Set(ControlMode::PercentOutput, 0.0); 
+        return;
+    }
+    else if(m_hood.GetSelectedSensorPosition() < ShooterConstants::hoodMin &&
+        m_angle <= ShooterConstants::hoodMin){
+        m_hood.Set(ControlMode::PercentOutput, 0.0);
+        return;
+    }
+    else {
+        m_hood.Set(ControlMode::Position, m_angle);
+    }
```

Comment:
I'd suggest refactoring this into:

```suggestion
    const double curr_hood_angle = m_hood.GetSelectedSensorPosition();
    const bool hood_overcurrent = m_hood.GetSupplyCurrent() >= ShooterConstants::zeroingcurrent;
    const bool hood_too_high = curr_hood_angle > ShooterConstants::hoodMax &&
                               m_angle >= ShooterConstants::hoodMax;
    const bool hood_too_low = curr_hood_angle < ShooterConstants::hoodMin &&
                                m_angle <= ShooterConstants::hoodMin;
                                                  
    if (hood_overcurrent || hood_too_high || hood_too_low) {
        m_hood.Set(ControlMode::PercentOutput, 0.0);
    } else {
        m_hood.Set(ControlMode::Position, m_angle);
    }
```

- make one call to `GetSelectedSensorPosition()` and cache the value in a local variable so you're not repeatedly hammering the CAN bus for the same data.
- remove the return statements in each branch since the cases are already mutually-exclusive. leaving them in might cause bugs later on if stuff is added after this code block (such as the smartdashboard call).